Python/ast.c

```diff
@@ -326,12 +326,234 @@ validate_expr(expr_ty exp, expr_context_ty ctx)
+                arg = asdl_seq_GET(values, i);
+                if (arg->kind == Starred_kind) {
+                    PyErr_SetString(PyExc_ValueError,
+                                    "can't use single star pattern inside of "
+                                    "a Call pattern");
+                    return 0;
+                }
+                if (!validate_pattern(arg)) {
+                    return 0;
+                }
+            }
+
+            asdl_keyword_seq *keywords = pattern->v.Call.keywords;
+            length = asdl_seq_LEN(keywords);
+            keyword_ty keyword;
+            for (i = 0; i < length; i++) {
+                keyword = asdl_seq_GET(keywords, i);
+                if (keyword->arg == Py_None) {
+                    PyErr_SetString(PyExc_ValueError,
+                                    "can't use double star pattern inside of "
+                                    "a Call pattern");
+                    return 0;
+                }
+                if (!validate_pattern(keyword->value)) {
+                    return 0;
+                }
+            }
+            return 1;
+        case Constant_kind:
+            // Ellipsis & Immutable sequences are not allowed
+            if (!validate_expr(pattern, Load)) {
+                return 0;
+            }
+            literal = pattern->v.Constant.value;
+            if (literal == Py_None || PyBool_Check(literal) ||
+                PyLong_CheckExact(literal) || PyFloat_CheckExact(literal) ||
+                PyBytes_CheckExact(literal) || PyComplex_CheckExact(literal) ||
+                PyUnicode_CheckExact(literal)) {
+                return 1;
+            }
+            PyErr_SetString(PyExc_ValueError,
+                            "unexpected constant inside of a literal pattern");
+            return 0;
+        case Dict_kind:
+            keys = pattern->v.Dict.keys;
+            values = pattern->v.Dict.values;
+            length = asdl_seq_LEN(keys);
+            if (length != asdl_seq_LEN(values)) {
+                PyErr_SetString(
+                    PyExc_ValueError,
+                    "Dict doesn't have the same number of keys as values");
+                return 0;
+            }
+            expr_ty key, value;
+            for (i = 0; i < length; i++) {
+                key = asdl_seq_GET(keys, i);
+                value = asdl_seq_GET(values, i);
+                if (!key) {
+                    // The position of **rest is handled on the compiler
+                    if (value->kind == Name_kind) {
+                        if (!validate_expr(value, Store)) {
+                            return 0;
+                        }
+                    }
+                    else {
+                        PyErr_SetString(PyExc_ValueError,
+                                        "the double star pattern's target can "
+                                        "only be Name");
+                        return 0;
+                    }
```

Comment:
This is a bit more readable for me:
```suggestion
                    if (value->kind != Name_kind) {
                        PyErr_SetString(PyExc_ValueError,
                                        "the double star pattern's target can "
                                        "only be Name");
                        return 0;
                    }
                    if (!validate_expr(value, Store)) {
                        return 0;
                    }
```